x/paychan/model.go

```diff
@@ -62,54 +61,13 @@ func (pc PaymentChannel) Copy() orm.CloneableData {
 	}
 }
 
-// PaymentChannelBucket is a wrapper over orm.Bucket that ensures that only
-// PaymentChannel entities can be persisted.
-type PaymentChannelBucket struct {
-	migration.Bucket
-	idSeq orm.Sequence
-}
-
 // NewPaymentChannelBucket returns a bucket for storing PaymentChannel state.
-func NewPaymentChannelBucket() PaymentChannelBucket {
-	b := migration.NewBucket("paychan", "paychan", orm.NewSimpleObj(nil, &PaymentChannel{}))
-	return PaymentChannelBucket{
-		Bucket: b,
-		idSeq:  b.Sequence("id"),
-	}
-}
-
-// Create adds given payment store entity to the store and returns the ID of
-// the newly inserted entity.
-func (b *PaymentChannelBucket) Create(db weave.KVStore, pc *PaymentChannel) (orm.Object, error) {
-	key, err := b.idSeq.NextVal(db)
-	if err != nil {
-		return nil, err
-	}
-	obj := orm.NewSimpleObj(key, pc)
-	return obj, b.Bucket.Save(db, obj)
+func NewPaymentChannelBucket() orm.ModelBucket {
+	b := newPaymentChannelObjectBucket()
+	return migration.NewModelBucket("paychan", orm.NewModelBucket(b))
```

Comment:
It would be nice to see an example of `orm.NewModelBucket` with indexes.

I guess you define one here: https://github.com/iov-one/weave/pull/615/files#diff-431dfabb3a7ef761096ec13914a0ee9dR142

But honestly, it looks a bit ugly. Maybe a nice helper to do that?
The construction of `orm.NewModelBucket` by just wrapping an existing Bucket...
Why not combine it?

Reply:
---

Done in https://github.com/iov-one/weave/pull/615/commits/8ee0a801a53b7dc2914c82aa642b7b33570d20b0